lib/cmd/fh3/app/logs/tail.js

```diff
@@ -32,20 +33,60 @@ module.exports = {
     return "api/v2/mbaas/" + fhc.curTarget + "/" + argv.deploytarget + "/apps/" + fhc.appId(argv.guid) + "/logs/chunk";
   },
   'preCmd': function (params, cb) {
-    var payload = {
+    var stdRequest = {
       "guid": params.app,
       "action": "logchunk",
       "deploytarget": params.env
     };
-    if (params.last) payload.last = params.last;
-    if (params.offset) payload.offset = params.offset;
-    if (params.logname) payload.logname = params.logname;
 
-    return cb(null, payload);
+    if (params.last) {
+      stdRequest.last = params.last;
+    }
+    if (params.offset) {
+      stdRequest.offset = params.offset;
+    }
+    if (params.logname) {
+      stdRequest.logname = params.logname;
+      return cb(null, stdRequest, null);
+    }else{
+      var listParams = {
+        "app": params.app,
+        "env": params.env
+      };
+      var self = this;
+      fhc.app.logs.list(listParams, function(err, res){
+        if(res.logs && res.logs.length > 1){
+          var stdLog = self.getFirst(res.logs, "stdout.log.");
+          var errorLog = self.getFirst(res.logs, "stderr.log.");
+          var errorRequest = _.clone(stdRequest);
+          stdRequest.logname = stdLog.name;
+          errorRequest.logname = errorLog.name;
+          log.warn("No log file specified. Fetching logs from latest log files. \nStandard output: " +
+            stdLog.name + "\nError log: " + errorLog.name);
+          return cb(null, stdRequest, errorRequest);
+        }else{
+          return cb(null, stdRequest, null);
```

Comment:
I think so. The _err_ from list endpoint is not handled here. Perhaps just fail the command if there is an error when listing logs.